kie-api/src/main/java/org/kie/api/builder/KieBuilder.java

```diff
@@ -39,6 +41,13 @@
     KieBuilder buildAll();
 
     /**
+     * Builds all the KieBases contained in the KieModule for which this KieBuilder has been created
+     * @param classFilters Used to prevent compilation of Java source files.
+     *          A Java source file vetoed by any of the given filters will not be compiled.
+     */
+    KieBuilder buildAll(Collection<KieClassFilter> classFilters);
```

Comment:
@mariofusco @mswiderski I had used a Collection because it made loading the filters in kie-wb-common quite simple, but I concede that's not a great reason to add this to the API. So I agree that only a single filter is needed as a parameter and will happily make that change. Similarly I have no issues with removing the KieClassFilter type and using a Predicate.

@psiroky I'm not sure that this feature will ever be needed for other types. As described, the issue is that (currently only in LiveSpark) some Java source files are meant to be GWT-compiled (and may even fail to Java compile). What this means is that, for the resource type `.java`, not every file with this extension is meant to be built the same way (with the Java compiler). I don't think this is true of any other resource type. For example, you always want Drools to build `.drl` files for you.